connectors/sources/servicenow.py

```diff
@@ -0,0 +1,346 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+"""ServiceNow source module responsible to fetch documents from ServiceNow."""
+import json
+import os
+
+import aiohttp
+from aiohttp.client_exceptions import ServerDisconnectedError
+
+from connectors.logger import logger
+from connectors.source import BaseDataSource, ConfigurableFieldValueError
+from connectors.utils import CancellableSleeps, iso_utc
+
+RETRIES = 3
+RETRY_INTERVAL = 2
+FETCH_SIZE = 1000
+MAX_CONCURRENCY_SUPPORT = 10
+
+RUNNING_FTEST = (
+    "RUNNING_FTEST" in os.environ
+)  # Flag to check if a connector is run for ftest or not.
+
+ENDPOINTS = {
+    "TABLE": "/api/now/table/{table}",
+    "ATTACHMENT": "/api/now/attachment",
+    "DOWNLOAD": "/api/now/attachment/{sys_id}/file",
```

Comment:
Question: I can't find these used anywhere. Will they be used in the future?

Reply:
Yes, It will be used for downloading attachment in `get_docs`